pkg/build/controller/strategy/util.go

```diff
@@ -90,3 +65,49 @@ func setupBuildEnv(build *buildapi.Build, pod *kapi.Pod) error {
 	}
 	return nil
 }
+
+// setupDockerSecrets mounts Docker Registry secrets into Pod running the build,
+// allowing Docker to authenticate against private registries or Docker Hub.
+func setupDockerSecrets(pod *kapi.Pod, secretRef *kapi.ObjectReference) {
+	if secretRef == nil {
+		return
+	}
+
+	// TODO: Should namespace be required here?
+	namespace := secretRef.Namespace
```

Comment:
@smarterclayton I'm concerned by this especially... the SecretVolumeSource require 'namespace' to be set... does that mean you have to provide namespace in the ObjectReference?